signing-and-verification-workflow.md

```diff
@@ -0,0 +1,53 @@
+# Signing and Verification Workflow
+
+This document describes how Notary v2 signs and verifies OCI artifacts.
+
+## Signing workflow
+
+The user wants to sign an OCI artifact and push the signature to a repository.
+
+### Prerequisite
+
+- User has access to the signing certificate and private key or a remote signing service through a notation plug-in.
+
+### Steps
+
+1. **Generate signature:** Using notation CLI or any other compliant signing tool, sign the OCI artifact. The signing tool should follow the following guideline.
+    1. Verify that the signing certificate is valid and satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
+    1. Verify that the signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
+    1. Generate signature.
+        1. Generate signature using signature formats specified in [supported signature envelopes](./signature-specification.md#supported-signature-envelopes).
+        1. Obtain an [RFC-3161](https://datatracker.ietf.org/doc/html/rfc3161.html) compliant timestamp for the signature generated in the previous step. This step is optional but recommended.
+            1. Verify that the timestamp signing certificate satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
+            1. Verify that the timestamp signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements).
+        1. Embed timestamp to the signature envelope.
+1. **Push the signature envelope:** Push the signature envelope generated in the previous step to the repository.
+1. **Generate signature artifact manifest:** As decribed in [signature specification](./signature-specification.md#storage) create the Notary v2 signature artifact manifest for the signature envelope generated in step 1.
+1. **Push signature artifact manifest:** Push Notary v2 signature artifact manifest to the repository.
+
+The user pushes the OCI artifact to the repository before the signature generation process as the signature reference must exist for the signature push to succeed. See: [ORAS Artifact Push Validation](https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest.md#push-validation) for more info.
+
+## Verification workflow
+
+The user wants to pull an OCI artifact only if they are signed by a trusted publisher and the signature is valid.
+
+### Prerequisites
+
+- User has a fully qualified reference to an OCI artifact they want to pull. If a tag is provided, the tag is looked up and converted to a digest.
```

Comment:
Notation will accept a url with tag and construct the descriptor from downloaded manifest, the registry maps tag to digest and send the artifact manifest as response.

Reply:
Because of security concerns around the value of resolved tag being different during signature verification and actual pull, for this PR we won't support automatic conversion of the tag to digest inside notary.